packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -45,26 +48,31 @@ export class AmountSelector extends BaseComponent {
 
   tokenBalanceController = new TokenBalanceController(this);
 
+  _handleOnResourceSelectedCallback = (): void => {
+    if (this.selectedResource) {
+      this.onResourceSelected(
+        this.selectedResource,
+        utils.parseUnits(
+          this.amount.toString(),
+          this.selectedResource.decimals ?? DEFAULT_ETH_DECIMALS
```

Comment:
I think is almost impossible for a resource to have undefined decimals. Because this info either is comming from `shared-config` or is going to come from another source of truth, where this is going to be hardcoded and defined. I'm not sure here if we need the `DEFAULT_ETH_DECIMALS` here 🤔

Reply:
Maybe it's true, but the interface from sygma-sdk-core tells that it could be undefined, so we keep the default value